mutt_signal.c

```diff
@@ -76,10 +76,12 @@ static void curses_signal_handler(int sig)
       /* We don't receive SIGWINCH when suspended; however, no harm is done by
        * just assuming we received one, and triggering the 'resize' anyway. */
       SigWinch = true;
+      notify_send(NeoMutt->notify_resize, NT_RESIZE, 0, NULL);
```

Comment:
doing that from signal context is a recipe for disaster.

the "easy" way out is to handle SigWinch more consistently, possibly posting the notification from there (you could centralize that in a check_signals() function invoked after each syscall which is a cancellation point).

the proper way is to use a self-pipe, which basically requires a proper event loop (the event loop itself can be had rather trivially, as you can see in https://sourceforge.net/p/isync/isync/ci/master/tree/src/util.c#l1090 - but fully converting the code base to an event driven architecture is anything but trivial).

Reply:
> doing that from signal context is a recipe for disaster.

argh!  I should have realised that.

> ... posting the notification from there ...
> ... after each syscall which is a cancellation point ...

`SIGWINCH` -\> `notify_send()` -\> sets flag on `RootWindow`
`SIGWINCH` -\> interrupts `getch()` -\> timeout notification
`RootWindow` observes timeout does reflow/repaint

That should be more robust.
 
> but fully converting the code base to an event driven architecture is anything but trivial

I know.  I've been trying for seven years now :-)
We _have_ made a lot of progress in that time, but there's still much to do.